src/Event.h

```diff
@@ -124,6 +195,26 @@ class EventMgr final : public Obj, public iosource::IOSource {
     uint64_t num_events_dispatched = 0;
 
 private:
+    /**
+     * Helper class to produce a compile time warning if Enqueue() is called with an explicit timestamp.
+     *
+     * All credits for this go to @bbannier.
+     *
+     * Remove in v8.1.
+     */
+    class DeprecatedTimestamp {
+    public:
+        DeprecatedTimestamp() : d(-1.0) {}
+        [[deprecated("Use overload Enqueue(EventMetadataVectorPtr meta, ...) to pass timestamp metadata")]]
+        DeprecatedTimestamp(double d) /* implicit */
+            : d(d) {}
+
+        explicit operator double() const { return d; }
+
+    private:
+        double d;
+    };
```

Comment:
Nit: Let's put `implicit` where one could put `explicit`. Also IMO no need to call out historical context (but thanks!).

```suggestion
    /**
     * Helper class to produce a compile time warning if Enqueue() is called with an explicit timestamp.
     *
     * Remove in v8.1.
     */
    class DeprecatedTimestamp {
    public:
        DeprecatedTimestamp() : d(-1.0) {}
        [[deprecated("Use overload EventMgr::Enqueue(EventMetadataVectorPtr meta, ...) to pass timestamp metadata")]]
        /*implicit*/ DeprecatedTimestamp(double d)
            : d(d) {}

        explicit operator double() const { return d; }

    private:
        double d;
    };
```